### ISSUE TITLE
Ampersand symbol in Piwik action String

### ISSUE THREAD

**Issue by maxmika** [matomo-org/piwik-sdk-ios#217]
We encountered a problem with Piwik action strings which contain the ampersand symbol (&).

## What we did:
When we send the Piwik event (let's say `"Imprint␣&␣Disclaimer"`) as a single one using the ...
``` Swift

sendSingleEventWithParameters:(NSDictionary*)parameters
                              success:(void (^)())successBlock
                              failure:(void (^)(BOOL shouldContinue))failureBlock {
    ...
}
```

... function, everything works fine. 

_______________

Now let's say a network error occurs. The event is saved and sent as a **bulk** together with the next Piwik event. 

``` Swift
sendBulkEventWithParameters:(NSDictionary*)parameters
                            success:(void (^)())successBlock
                            failure:(void (^)(BOOL shouldContinue))failureBlock {
```

In this case, both event parameters are sent in the body of the `NSURLMutableRequest`.

## The Problem:
What we receive in our Piwik backend system for the first string is only `Imprint␣`, thus not matching with  `Imprint␣&␣Disclaimer`. That means, the `&` symbol is interpreted as a URL Parameter separator.

## What we tried
is to escape all ampersand signs with `%26` for events sent as bulk and with `&amp;` for request with just a single event. Doing so, we could at least achieve a `&amp;` in both cases in our backend, but we had to modify the Piwik SDK framework code.

## Question
Is there a way to send actions containing an ampersand symbol without modifying the SDK's code? What do you suggest?

Thanks for your help in advance!

**Comment by brototyp** [comment]
Hi @maxmika, which version of the SDK are you using?

**Comment by brototyp** [comment]
I guess you are using an older (< 4.0) version of the SDK. Please consider upgrading to the latest version.

**Comment by brototyp** [comment]
Nevertheless. The issue is still in the newest version. I will dig into it and see if I can find a solution.

**Comment by maxmika** [comment]
@brototyp Your fix works just fine for us – I would be grateful if you could apply it to the CocoaPods Version asap!

Thank you very much fur your quick solution!

**Comment by brototyp** [comment]
Hi @maxmika, thanks for checking the changes. I just merged it. It will be part of the next release.

**Comment by maxmika** [comment]
Thank you, @brototyp . 
Can you already tell when the next release will be published (approximately)?

**Comment by brototyp** [comment]
Hi @maxmika, I just released 4.4.2 containing this fix.
